### PR TITLE
WIP: Add support for fallback page, webroot. fix issue with directory listing.

### PR DESCRIPTION
**Pull request by ylebre**

### CODE REVIEW

File: lib/apis/archive-files.js [hunk 1]
```diff
@@ -93,13 +93,28 @@ module.exports = class ArchiveFilesAPI {
...
     if (!filepath) filepath = '/'
     var isFolder = filepath.endsWith('/')
     var entry
     const tryStat = async (path) => {
       if (entry) return
+      // apply the web_root config
+      if (manifest && manifest.web_root) {
+        if (path) {
+        //   path = joinPaths(manifest.web_root, path)
+          path = manifest.web_root + path
```
Comment (ylebre) [inline lib/apis/archive-files.js]: This specific line used the 'joinPaths' function from path, but I wasn't sure how to import that function here.

**Comment by pfrazee** [comment]
I'll do some point-by-point review, but (I hate to say it) we really need to split this PR into one PR for each issue. Can you pull this apart so that it's one PR for fallback page, one PR for webroot, and one PR for custom domains?

File: lib/apis/archive-files.js (hunk 1 above)
Comment (pfrazee) [reply lib/apis/archive-files.js]: You can do `const joinPath = require('path').join` in the top of the file

**Comment by ylebre** [comment]
Sure thing, I'll see if I can whip those up.

**Pull request closed without merging**